compiler/hash-typecheck/src/ops/validate.rs

```diff
@@ -68,6 +80,204 @@ impl<'gs, 'ls, 'cd> Validator<'gs, 'ls, 'cd> {
         todo!()
     }
 
+    /// Ensure the element `merge_element_term_id` of the merge with the given `merge_term_id` is
+    /// either level 2 (along with the merge being all level 2), or level 1 (along with the merge
+    /// being all level 1). Furthermore, if it is level 1, the merge should only have zero or one
+    /// nominal definition attached.
+    fn validate_merge_element(
+        &mut self,
+        merge_kind: &mut MergeKind,
+        merge_term_id: TermId,
+        merge_element_term_id: TermId,
+    ) -> TcResult<()> {
+        let reader = self.reader();
+        let merge_element_term = reader.get_term(merge_element_term_id);
+
+        // Error helper:
+        let invalid_merge_element = || -> TcResult<()> {
+            Err(TcError::InvalidElementOfMerge {
+                term: merge_element_term_id,
+            })
+        };
+
+        // Helper to ensure that a merge is level 2, returns the updated MergeKind.
+        let ensure_merge_is_level2 = || -> TcResult<MergeKind> {
```

Comment:
These are not standalone functions because 1) they are not used elsewere, and 2) they capture local variables "merge_kind", "merge_term_id", and "merge_element_term_id". Having them as standalone functions might be a bit too clunky and wouldn't provide much advantage in my view. What do you think?

Reply:
Yeah I agree, if they aren’t being used anywhere else, completely fine to leave them as closures.